addons/mail/static/src/models/thread/thread_tests.js

```diff
@@ -155,6 +157,144 @@ QUnit.test('create (chat)', async function (assert) {
+        assert.strictEqual(
+            action.type,
+            'ir.actions.act_window',
+            "action should be the one to open view"
+        );
+        assert.strictEqual(
+            action.res_id,
+            this.env.messaging.currentPartner.id,
+            "action should have id of current partner"
+        );
+        assert.strictEqual(
+            action.res_model,
+            'res.partner',
+            "action should have 'res.partner' model"
+        );
+    });
+    await this.start({ bus });
+
+    await this.env.models['mail.thread'].openChat({
+        partnerId: this.env.messaging.currentPartner.id,
+    });
+    assert.verifySteps(
+        ['do_action'],
+        "should have done an action to open form view"
+    );
+});
+
+QUnit.test('openChat: open form view for non user', async function (assert) {
+    assert.expect(5);
+
+    const bus = new Bus();
+    bus.on('do-action', null, ({ action }) => {
+        assert.step('do_action');
+        assert.strictEqual(
+            action.type,
+            'ir.actions.act_window',
+            "action should be the one to open view"
+        );
+        assert.strictEqual(
+            action.res_id,
+            14,
+            "action should have id of target partner"
+        );
+        assert.strictEqual(
+            action.res_model,
+            'res.partner',
+            "action should have 'res.partner' model"
+        );
+    });
+    await this.start({ bus });
+
+    await this.env.models['mail.thread'].openChat({
+        partnerId: 14, // random id that is not current partner
+    });
+    assert.verifySteps(
+        ['do_action'],
+        "should have done an action to open form view"
+    );
+});
+
+QUnit.test('openChat: open new chat for user', async function (assert) {
+    assert.expect(3);
+
+    this.data['res.users'].records = [{
+        id: 11,
+        partner_id: 14, // random id that is not current partner
+    }];
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'channel_get') {
+                return {
+                    channel_type: 'chat',
+                    direct_partner: [{
+                        id: 14,
+                    }],
+                    id: 47,
+                    is_pinned: true,
+                };
+            }
+            return this._super(...arguments);
+        },
+    });
+
+    await this.env.models['mail.thread'].openChat({
+        partnerId: 14,
+    });
+    const chat = this.env.models['mail.thread'].find(thread =>
+        thread.model === 'mail.channel' &&
+        thread.channel_type === 'chat' &&
+        thread.correspondent &&
+        thread.correspondent.id === 14
+    );
+    assert.ok(chat, 'a chat should exist with the target partner');
+    assert.strictEqual(chat.id, 47, 'the chat should be a new chat');
+    assert.strictEqual(chat.viewers.length, 1, 'the chat should be displayed in a thread viewer');
```

Comment:
This is nice!

Just small changes to make test even better: assertion before the `openChat()` (whether it's chat does not exist, or chat exist but has no viewers yet, etc.)

Reply:
todo